Userland/Libraries/LibGfx/PNGWriter.cpp

```diff
@@ -174,35 +185,37 @@ void PNGWriter::add_IDAT_chunk(Gfx::Bitmap const& bitmap)
             ByteBuffer buffer {};
             int sum = 0;
 
-            void append(u8 byte)
+            ErrorOr<void> append(u8 byte)
             {
-                buffer.append(byte);
+                TRY(buffer.try_append(byte));
                 sum += static_cast<i8>(byte);
+                return {};
             }
 
-            void append(AK::SIMD::u8x4 simd)
+            ErrorOr<void> append(AK::SIMD::u8x4 simd)
             {
-                append(simd[0]);
-                append(simd[1]);
-                append(simd[2]);
-                append(simd[3]);
+                TRY(append(simd[0]));
+                TRY(append(simd[1]));
+                TRY(append(simd[2]));
+                TRY(append(simd[3]));
+                return {};
             }
         };
 
         Filter none_filter { .type = PNG::FilterType::None };
-        none_filter.buffer.ensure_capacity(sizeof(Pixel) * bitmap.height());
+        TRY(none_filter.buffer.try_ensure_capacity(sizeof(Pixel) * bitmap.height()));
```

Comment:
Should the filter buffer sizes not be bitmap.width instead of height?
```suggestion
        TRY(none_filter.buffer.try_ensure_capacity(sizeof(Pixel) * bitmap.width()));
```